### ISSUE TITLE
Controller mapping for specific system in MAIN MENU

### ISSUE THREAD

**Issue by azvarel** [recalbox/recalbox-os#1228]
Following manual https://github.com/recalbox/recalbox-os/wiki/Custom-controller-mapping-for-specific-system-(EN) to mapping the buttons gets quite annoying to do one by one, could add this in the option
**MAIN MENU > GAMES SETTINGS > ADVANCED > within each system**, (could place the mapping in the same style as the mapping of the buttons when the controllers are paired.)

**Comment by substring** [comment]
Well, to be honnest, i'm afraid this is really too much work for such a feature if you expect to remapp your pad for each system, then have recalbox force the mapping file for the required system.

Pretty much work, indeed ... + consider using a custom config file is not always about just inputs, so it also means that the auto configuration should handle that too ...

Yeah, really too much work according to the way i understand your suggestion

**Comment by azvarel** [comment]
The configuration is only for emulators. I'll give you an example, I use xbox360 controls when playing Super Mario Bros. of the NES, (in the original control NES: hold B it runs, press A it jumps), being the hands ache, by the position is more pleasant to put the button B being the button X and the button A in the button A of the X360, I know if you understand or are confused?

![mapper button](https://cloud.githubusercontent.com/assets/129056/23762702/a19d443c-04d5-11e7-943e-39cdabc17fcb.png)

**Comment by substring** [comment]
Have ypu tried core rempas in retroarch ?

**Comment by azvarel** [comment]
Yes, it's as I described in the first text, it's annoying to do that over there, I have to get into the game, enter the hotkey + b, enter the core emulator> controls and edit there, this one by one, Much faster and agile because it is a single screen would have all the emulators to be adjusted their mappings

**Comment by daaa57150** [comment]
I'm also for the "anatomically better position", all 2 button systems have that annoyance.
I'm not sure we need a special mapping for each system though, just use the snes YB instead of BA for those in the auto config. Is that easy enough?

**Comment by azvarel** [comment]
@daaa57150 No, it's not just with 2 buttons, there are other specials example games that require 6 buttons, example Super Street Fighter II of CPS1 where the order of the buttons being based on SNES would be L (high punch) change by X (medium punch) Same thing for R (high kick) change by A (medium kick)

**Comment by daaa57150** [comment]
@azvarel Sorry I was talking specifically about systems with 2 buttons. I know arcade games are a pain to configure, I tried and just gave up, you can't find a unique mapping layout that works with all arcade games, they have to be configured individually in the mame menu (R2).

@substring How hard would it be to adopt that "anatomically better position" for systems with 2 buttons?

**Comment by paradadf** [comment]
@daa5715 you are assuming that everyone uses a snes pad like controller layout, but there are many others that are used (nes pad, arcade, genesis etc)  and are already "ergonomically better". IMO, if you want to change it for your specific controller, just do it from retroarch and everyone is happy.

**Comment by daaa57150** [comment]
@paradadf you're right, good point, I didn't think about that.

**Comment by azvarel** [comment]
@paradadf 

![controles-videogames-raros-marcantes_27](https://cloud.githubusercontent.com/assets/129056/24401589/5d6640a6-138b-11e7-9975-f96bea9d1938.jpg)

You know that this is not true, there are innumerable situations, the same neogeo control is C + D + A + B where C (high kick) A (low kick) B (low kick) and I like to change For C + A (to be punched) and D + B (to be kicked), already in controls arcades this configuration changes, and we do not use this pattern of order, we use this here:

![simple4](https://cloud.githubusercontent.com/assets/129056/24402119/8fabecf8-138d-11e7-8d8f-8c5a3fe9e519.jpg)

This is totally cultural, in the region where I grew up playing arcade all the places had this setting for neogeo, for cps the top buttons were standard all buttons up punches, and all buttons down kicks.

**Comment by paradadf** [comment]
@azvarel I don't understand what wasn't true and can't find where my comment conflicts with your last message. What you just said is exactly what I said before about the controller layout. Or am I missing something?

**Comment by paradadf** [comment]
@azvarel you tagged me. But no need to apologize.

**Comment by daaa57150** [comment]
@azvarel 
Just to recap, you are saying you want to have a specific configuration for each emulator, right? 
But then you tell me that neogeo has X or Y pattern whereas CPS1 has another. They run in the same emulator, one configuration won't work for both + every other arcade board. 
That's all I'm saying.

**Comment by azvarel** [comment]
@daaa57150 of course it gives, even though it is the same emulator, there is a .cfg file that you can do this

**Comment by gYonder** [comment]
I've been trying to do this specific remapping according to https://github.com/recalbox/recalbox-os/wiki/Custom-controller-mapping-for-specific-system-(EN)
Some systems have the core remap feature, and some don't (like NES). No matter in what ways I try, I can't get this to save. The cfg files mentioned in this guide do not exist, even after choosing save new configuration (4.1.0). Can somebody explain to me how to make this remapping stay, per system?

**Comment by TigerJon** [comment]
The tutorial in https://github.com/recalbox/recalbox-os/wiki/Custom-controller-mapping-for-specific-system-(EN) is right BUT the paths are totally wrong. The right ones are:

Retroarch will save the libretro.cfg file (commonly fba.libretro.cfg) at:
 
/recalbox/share/system/.config/retroarch/config (The .config folder is a hidden folder btw)

You will put the custom cfg file (using neogeo as an example, rename the fba_libretro.cfg to neogeocustom.cfg) in /recalbox/share/system/configs/retroarch/inputs

And finally, you will add the following line at recalbox.conf  (if you are using windows, use notepad++ to edit the file) : 

neogeo.configfile=/recalbox/share/system/configs/retroarch/inputs/neogeocustom.cfg

It works fine at both recalbox and batocera. I still need to test if this works with another systems like cps1, cps2, etc.

**Comment by substring** [comment]
@TigerJon a path is a path. Forcing a path in the .configfile option to something different than what we say in the wiki would lead you to some mistakes. The paths are right unless you had already changed the default behaviour before.

**Comment by TigerJon** [comment]
@substring everything is default here, tried lots of time with fresh installs (Recalbox and Batocera) and using the paths described in the article makes no effect because they are wrong. Retroarch does not save the custom config in the mentioned path but does save in the /.config/retroarch one. I suggest you to try the same in the article to see that it DOES NOT WORK. The user @azvarel tried and did not worked, I did tried and did not worked. But I payed attention to see in which folder Retroarch saves the file and now IT WORKS :). Why don't you try too?

**Comment by substring** [comment]
@TigerJon

**Comment by substring** [comment]
damn misclicked. Will have a look at that, it may seem you are right
